docs/modules/ROOT/pages/tutorials/migration.adoc

```diff
@@ -0,0 +1,645 @@
+              key: token
+        - name: K8S_NAMESPACE
+          value: rsync-test
+        volumeMounts:
+        - mountPath: /rsync-source
+          name: rsync-source
+      volumes:
+      - name: rsync-source
+        persistentVolumeClaim:
+          claimName: rsync-source
+--
+<1> Please refer to https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/[the Kubernetes documentation] on common labels.
+
+
+IMPORTANT: Be aware that `oc rsync` has different options than `rsync` itself.
+
+[source,bash]
+--
+Options:
+     --compress=false: compress file data during the transfer
+ -c, --container='': Container within the pod
+     --delete=false: If true, delete files not present in source
+     --exclude=[]: When specified, exclude files matching pattern
+     --include=[]: When specified, include files matching pattern
+     --no-perms=false: If true, do not transfer permissions
+     --progress=false: If true, show progress during transfer
+ -q, --quiet=false: Suppress non-error messages
+     --strategy='': Specify which strategy to use for copy: rsync, rsync-daemon, or tar
+ -w, --watch=false: Watch directory for changes and resync automatically
+--
+
+=== File Integrity Check
+
+After the migration, you should check the integrity of your data with the following commands:
+
+.Calculate the checksum for all files at the origin
+[source,bash]
+--
+$ find . -type d -exec sh -c "cd '{}' && find . -maxdepth 1 -type f ! -name COPYSHA1SUMS -printf '%P\0' | xargs -r0 sha1sum -- > COPYSHA1SUMS" \;
+--
+
+.Create a log with a verification of all files at the destination
+[source,bash]
+--
+$ cd <path> && find . -type d -exec sh -c "cd '{}' && echo '{}' && sha1sum -c COPYSHA1SUMS" \; > sha1sums-verify-log-$(date +%F-%H-%M).log 2>&1
+--
+
+== Container Images
+
+Since APPUiO Cloud is based on OpenShift 4, there are new requirements for your container images. This section contains all the required steps for adapting your images to the new environment.
+
+=== On OpenShift 3
+
+TIP: This section uses the https://github.com/containers/skopeo[skopeo] tool for managing images and repositories.
+
+. Get a user token at this URL: https://<origin-cluster-console>/oauth/token/request.
+
+. Use the generated user token to authenticate to the registry on the command line. As the user token has enough privileges to read the image, a service account token isn't required.
++
+[source,bash]
+--
+$ skopeo login -u openshift -p <token> <origin-url>
+--
++
+Skopeo uses the docker auth config. So this should look like:
++
+[source,bash]
+--
+$ cat ~/.docker/config.json
+{
+    "auths": {
+        "<origin-url>": {
+            "auth": "...="
+        }
+    }
+}
+--
++
+Check if the access is working:
++
+[source,bash]
+--
+$ skopeo inspect docker://<origin-url>/<namespace>/<image>:<image-tag>
+--
+
+IMPORTANT: This is a user token, and therefore it expires when you logout.
+
+=== On OpenShift 4
```

Comment:
This part might need APUB-72 which is scheduled for the next sprint (aka next week).

Reply:
Should I add a reference to this, or just leave it like that, since it should be something available by the time we publish this?

Reply:
No ref needed, just leave it as it is.

Reply:
Perfect! Resolving this conversation.